java/compaction/compaction-job-creation/src/test/java/sleeper/compaction/job/creation/CreateCompactionJobsTest.java

```diff
@@ -304,6 +305,37 @@ void shouldCreateJobsWhenStrategyDoesNotCreateJobsForSplitFilesWhenCompactingAll
                 verifyJobCreationReported(job);
             });
         }
+
+        @Test
+        void shouldCreateJobsLimitedDownToExecutionLimitWhenTheCompactionJobsExceedTheValue() throws Exception {
+
+            //Given normal compactation we set a limit for the execution to be less than the entries present 
+            tableProperties.set(COMPACTION_STRATEGY_CLASS, BasicCompactionStrategy.class.getName());
+            tableProperties.set(COMPACTION_FILES_BATCH_SIZE, "3");
+            instanceProperties.set(COMPACTION_JOB_EXECUTION_LIMIT, "150");
+            stateStore.initialise(new PartitionsBuilder(schema).singlePartition("root").buildList());
+            FileReferenceFactory factory = FileReferenceFactory.fromUpdatedAt(stateStore, DEFAULT_UPDATE_TIME);
+
+            FileReference fileReference1 = factory.rootFile("file1", 200L);
+            stateStore.addFiles(List.of(fileReference1));
+
+            // When we force create jobs
+            createJobs(Mode.FORCE_ALL_FILES_AFTER_STRATEGY);
```

Comment:
This is a good start, but it's only creating one job because there's only one file. We could also do with a test where the limit is exceeded without forcing all files to be compacted.